pages/latest/[urlTitle].tsx

```diff
@@ -41,20 +45,59 @@ const componentsUI: MDXProviderComponentsProp = {
     a: LinkMarkdown
 };
 
-const UrlTitle = ({ data, source, error }: PageData): JSX.Element => {
+const UrlTitle = ({ data, source, toc, error }: PageData): JSX.Element => {
     const content = source ? hydrate(source, { components: componentsUI }) : null;
+    // ---- Table Of Content Active Item
+    const [tocActive, setTocActive] = useState(null);
+    const observer = useRef(null);
+    const options = {
+        root: null,
+        rootMargin: '0px 0px -70% 0px',
+        threshold: 1
+    };
+    useEffect(() => {
+        const targets = document.querySelectorAll('h2,h3');
+        if (observer.current) {
+            observer.current.disconnect();
+        }
+        observer.current = new IntersectionObserver(([entry]) => {
+            if (entry.isIntersecting) {
+                setTocActive(entry.target.id);
+            }
+        }, options);
+        const { current: currentObserver } = observer;
+        targets.forEach((target) => currentObserver.observe(target));
+        return () => currentObserver.disconnect();
+    });
     return (
         <>
             <Head>
                 <title>{data.title}</title>
             </Head>
-            <h1>{data.title}</h1>
             {error ? (
-                <Terminal content={error} />
+                <main className="container mt-4 md:mt-0 justify-self-center overflow-auto">
```

Comment:
I think we can extract these classes to a more descriptive class and use Tailwind's `@apply`